src/main/java/com/google/graphgeckos/dashboard/datatypes/GitHubData.java

```diff
@@ -0,0 +1,84 @@
+package com.google.graphgeckos.dashboard.datatypes;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.cloud.Timestamp;
+import java.util.Date;
+import java.util.Map;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class GitHubData {
+
+  private String branch;
+
+  private String commitHash;
+
+  private Timestamp timestamp;
+
+  private String repositoryLink;
+
+  /**
+   * Extracts commit id (also known as commit hash) and timestamp nested fields from the data of the head_commit
+   * field.
+   * @param headCommit representation of the head_commit field of the GitHub API request json.
+   */
+  @JsonProperty("head_commit")
+  private void unpackHeadCommit(Map<String, Object> headCommit) {
+    commitHash = headCommit.get("id").toString();
+    timestamp = Timestamp.of(new Date(headCommit.get("timestamp").toString()));
```

Comment:
how can we test that this custom code (and the other codes annotated with `@JsonProperty`) works as expected?
One way would be to have this not private but default and have a test case that executes this and checks the values of commitHash and timestamp of this object. There may be better ways to test this in Spring.

Reply:
I tested similar POJO using public getters.

Reply:
Could you add tests like those to this PR?

Reply:
Done!